lib/collection/property-base.js

```diff
@@ -60,6 +60,28 @@ PropertyBase = function PropertyBase (definition) {
 _.assign(PropertyBase.prototype, /** @lends PropertyBase.prototype */ {
 
     /**
+     * Invokes the given iterator for every parent in the parent chain of the given element.
+     *
+     * @param {?Object|Boolean} [options={}] - A set of options for the parent chain traversal.
+     * @param {?Boolean} [options.withRoot=false] - Set to true to include the collection object as well.
+     * @param {Function} iterator - The function to call for every parent in the ancestry chain.
+     * @todo Cache the results
+     */
+    forEachParent: function (options, iterator) {
+        _.isFunction(options) && (iterator = options, options = {});
+        if (!_.isFunction(iterator)) { return; }
+        if (!_.isObject(options)) { options = { withRoot: options }; }
+
+        var parent = this.parent(),
+            grandparent = parent && parent.parent();
```

Comment:
safety is paramount, we _never_ throw errors in the SDK, so add extra checks to ensure that `.parent` is a function